sdk/tables/Azure.Data.Tables/src/TableClient.cs

```diff
@@ -450,12 +450,147 @@ public virtual Response Update(IDictionary<string, object> entity, string eTag,
         /// Merges the specified table entity by updating only the properties present in the supplied entity, if it exists. Inserts the entity if it does not exist.
         /// </summary>
         /// <param name="entity">The entity to merge.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns>The <see cref="Response"/> indicating the result of the operation.</returns>
+        public virtual async Task<Response> UpsertMergeAsync(IDictionary<string, object> entity, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(entity, nameof(entity));
+
+            //TODO: Create Resource strings
+            if (!entity.TryGetValue(TableConstants.PropertyNames.PartitionKey, out var partitionKey))
+            {
+                throw new ArgumentException("The entity must contain a PartitionKey value", nameof(entity));
```

Comment:
Since we have a few appearances of these exception strings now, should we create a constant?